commit 25e68494fad3
Author: Jan-Ivar Bruaroey <jib@mozilla.com>
Date:   Fri Dec 1 20:49:46 2017 -0500

    Bug 1422389 - Limit default engine resolution. r=mchiang, a=abillings
    
    MozReview-Commit-ID: LS1z7I4myeb
    
    --HG--
    extra : rebase_source : 200216163b4555ede8912e64871eea81eb8affac
    extra : source : f0fd6e20a13d11660a5ac57e9703f09978f12acb
---
 dom/media/webrtc/MediaEngineDefault.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git dom/media/webrtc/MediaEngineDefault.cpp dom/media/webrtc/MediaEngineDefault.cpp
index ea5996e92e23..01efb04b5df8 100644
--- dom/media/webrtc/MediaEngineDefault.cpp
+++ dom/media/webrtc/MediaEngineDefault.cpp
@@ -120,6 +120,8 @@ MediaEngineDefaultVideoSource::Allocate(const dom::MediaTrackConstraints &aConst
                                 MediaEngine::DEFAULT_43_VIDEO_HEIGHT
 #endif
                                 );
+  mOpts.mWidth = std::max(160, std::min(mOpts.mWidth, 4096));
+  mOpts.mHeight = std::max(90, std::min(mOpts.mHeight, 2160));
   mState = kAllocated;
   *aOutHandle = nullptr;
   return NS_OK;
